core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -745,17 +727,12 @@ private[zio] final class FiberContext[E, A](
       // interrupted, but do so using a weak finalizer, which will be removed
       // as soon as the key is garbage collected:
       val key = parentScope.unsafeEnsure(
-        exit =>
+        _ =>
           UIO.effectSuspendTotal {
             val childContext = childContextRef()
 
             if (childContext ne null) {
-              currentSuperviseMode match {
-                case ForkSuperviseMode.Auto =>
-                  if (exit.interrupted) childContext.interruptAs(fiberId) else childContext.await
-                case ForkSuperviseMode.Interrupt => childContext.interruptAs(fiberId)
-                case ForkSuperviseMode.Await     => childContext.await
-              }
+              childContext.interruptAs(fiberId)
```

Comment:
Ah crap, we are interrupting as the wrong fiber id. Maybe we do need `Exit[Any, Any]`??? Because then we can extract the correct interrupting fiber id if it exists inside the `Exit`.